specs/Print.md

```diff
@@ -0,0 +1,473 @@
+  [propget] HRESULT QualityVertical([out, retval] INT32* value);
+
+  /// Set the `QualityVertical` property.
+  [propput] HRESULT QualityVertical([in] INT32 value);
+
+  /// The name of the printer to use. Defaults to empty string.
+  /// If the printer name is empty string or null, then it prints to the default printer on the OS.
+  /// This value is ignored in PrintToPdfStream method.
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`. See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings)
+  [propget] HRESULT PrinterName([out, retval] LPWSTR* value);
+
+  /// Set the `PrinterName` property. If provided printer name doesn't match with the name of any printers on the OS,
+  /// `ICoreWebView2PrintWithSettingsCompletedHandler` handler will return `E_ABORT`.
+  [propput] HRESULT PrinterName([in] LPCWSTR value);
+}
+```
+
+```c# (but really MIDL3)
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2Environment;
+    runtimeclass CoreWebView2PrintSettings;
+    runtimeclass CoreWebView2;
+
+    enum CoreWebView2PrintSide
+    {
+        Single = 0,
+        BothHorizontal = 1,
+        BothVertical = 2,
+    };
+
+    enum CoreWebView2PrintDialogKind
+    {
+        BrowserPrintPreview = 0,
+        SystemPrint = 1,
+    };
+
+    runtimeclass CoreWebView2PrintSettings
+    {
+        // ICoreWebView2PrintSettings members
+        CoreWebView2PrintOrientation Orientation { get; set; };
+        Double ScaleFactor { get; set; };
+        Double PageWidth { get; set; };
+        Double PageHeight { get; set; };
+        Double MarginTop { get; set; };
+        Double MarginBottom { get; set; };
+        Double MarginLeft { get; set; };
+        Double MarginRight { get; set; };
+        Boolean ShouldPrintBackgrounds { get; set; };
+        Boolean ShouldPrintSelectionOnly { get; set; };
+        Boolean ShouldPrintHeaderAndFooter { get; set; };
+        String HeaderTitle { get; set; };
+        String FooterUri { get; set; };
+
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2PrintSettings2")]
+        {
+            // ICoreWebView2PrintSettings2 members
+            String PageRanges { get; set; };
+            Int32 PagesPerSheet { get; set; };
+            Int32 Copies { get; set; };
+            Boolean ShouldCollate { get; set; };
+            Boolean IsColor { get; set; };
+            CoreWebView2PrintSide PrintSide { get; set; };
+            Int32 QualityHorizontal { get; set; };
+            Int32 QualityVertical { get; set; };
+            String PrinterName { get; set; };
+        }
+    }
+
+    runtimeclass CoreWebView2
+    {
+        // ...
+
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2_15")]
+        {
+            void Print(CoreWebView2PrintDialogKind printDialogKind);
+            void Print();
+
+            Windows.Foundation.IAsyncAction PrintWithSettingsAsync(CoreWebView2PrintSettings printSettings);
```

Comment:
Does this do the browser or system print?

Reply:
i think this api wouldn't show ui ?